src/main/java/de/zalando/aruha/nakadi/repository/kafka/KafkaLocationManager.java

```diff
@@ -0,0 +1,129 @@
+package de.zalando.aruha.nakadi.repository.kafka;
+
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.Properties;
+
+import javax.annotation.PostConstruct;
+
+import org.apache.curator.framework.CuratorFramework;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import de.zalando.aruha.nakadi.repository.zookeeper.ZooKeeperHolder;
+
+public class KafkaLocationManager {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaLocationManager.class);
+
+    private final String _BROKERS_IDS_PATH = "/brokers/ids";
+
+    @Autowired
+    private ZooKeeperHolder zkFactory;
+
+    private Brokers kafkaBrokerList;
+
+    private Properties kafkaProperties;
+
+    class Broker implements Comparable<Broker> {
+        final String host;
+        final Integer port;
+
+        public Broker(final String host, final Integer port) {
+            this.host = host;
+            this.port = port;
+        }
+
+        @Override
+        public int compareTo(Broker other) {
+            final int result = host.compareTo(other.host);
+            return result == 0 ? port.compareTo(other.port) : result;
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            if (other == this)
+                return true;
+            if (!(other instanceof Broker))
+                return false;
+            return this.compareTo((Broker)other) == 0;
+        }
+
+        public String toString() {
+            return this.host + ":" + this.port;
+        }
+    }
+
+    class Brokers extends ArrayList<Broker> {
```

Comment:
I also think that this class can be removed now. It was created as a fix for DependencyInjection. But now the list of brokers is not a bean and not injected anywhere, so I think we don't need Brokers class any more.